aspnetcore/diagnostics/asp0015.md

```diff
@@ -0,0 +1,32 @@
+---
+title: "ASP0015: Suggest using IHeaderDictionary properties"
+description: "Learn about analysis rule ASP0015: Suggest using IHeaderDictionary properties"
+author: tdykstra
+monikerRange: '>= aspnetcore-7.0'
+ms.author: tdykstra
+ms.date: 11/22/2022
+uid: diagnostics/asp0015
+---
+# ASP0015: Suggest using IHeaderDictionary properties
+
+| | Value |
+|-|-|
+| **Rule ID** |ASP0015|
+| **Category** |Usage|
+| **Fix is breaking or non-breaking** |Non-breaking|
+
+## Cause
+
+[IHeaderDictionary](xref:System.Collections.IDictionary) properties are the recommended strategy for accessing headers.
+
+## Rule description
+
+`IHeaderDictionary` properties are recommended for accessing headers.
```

Comment:
```suggestion
`IHeaderDictionary` properties are recommended for accessing headers. Accessing headers using an indexer as in the example below is not recommended.

```csharp
var app = WebApplication.Create();

app.MapGet("/", (HttpContext context) => context.Request.Headers[""content-type""]);

app.Run();
```

```

The formatting on the suggestion is a little off because GitHub's suggestion UI doesn't play nice with codeblocks inside suggestions.

Reply:
I fixed that by using triple-tildes for the suggestion. [Inline code blocks](https://review.learn.microsoft.com/en-us/help/platform/code-in-docs?branch=main#inline-code-blocks)